tools/common_lib/src/gemm.cpp

```diff
@@ -12,14 +12,32 @@ std::vector<std::byte> dnnl_gemm_op::gemm(const bindings_t& bindings, opts_t opt
 
     dnnl::memory input_a_memory = [&](const auto& binding)
     {
-        auto ret = dnnl::memory(to_dnnl_mem_desc(binding.shape, binding.layout, binding.dt), engine);
+        dnnl::memory ret;
+        if (binding.transposed)
```

Comment:
As mentioned in another comments, please remove transposed flag from bindings. You should just pass DataLayout and use in this function.

Reply:
As per our discussion, I have moved the transposed flag to the opts struct in the dnnl_gemm_opts namespace.